app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java

```diff
@@ -254,20 +297,34 @@ private void setFavorActivatedView(View v) {
     cancelFavorBtn.setVisibility(View.VISIBLE);
     chatBtn.setVisibility(View.VISIBLE);
     toggleTextViewsEditable(false);
-    updateViewFromStatus(v);
+    updateViewFromStatus();
   }
 
   /** When edit button is clicked */
   private void startUpdatingFavor() {
-    View rootView = getView();
     favorStatus = FavorStatus.EDIT;
-    updateViewFromStatus(rootView);
+    updateViewFromStatus();
+  }
+
+  private void completeFavor() {
+    if (currentFavor.getStatusId() == FavorStatus.ACCEPTED.toInt()) {
```

Comment:
What if the favorstatus is cancelled?

Reply:
There is another "if" to catch these cases before calling 
```
 if (favorStatus.equals(FavorStatus.ACCEPTED)
              || favorStatus.equals(FavorStatus.COMPLETED_ACCEPTER)) {
            completeFavor();
          } 
```